cardano-testnet/src/Testnet/Start/Types.hs

```diff
@@ -1,23 +1,34 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE NumericUnderscores #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
 module Testnet.Start.Types
   ( CardanoTestnetCliOptions(..)
   , CardanoTestnetOptions(..)
+  , NumDReps(..)
+  , NumPools(..)
+  , NumRelays(..)
+  , cardanoNumPools
+  , cardanoNumRelays
 
   , anyEraToString
   , anyShelleyBasedEraToString
   , eraToString
 
   , TestnetNodeOptions(..)
-  , extraSpoNodeCliArgs
+  , testnetNodeCfgFile
+  , testnetNodeExtraCliArgs
+  , isSpoNodeOptions
+  , isRelayNodeOptions
```

Comment:
Personally I am not a big fan of exporting small helper functions like this. Users could just pattern match on the constructors, instead of using those functions.

Optional.

Reply:
Those functions are more readable than pattern matching on the constructor. This is obvious to me
```haskell
when (isSpoNodeOptions nodeOptions) $ do
  ...     
```
rather than more noisy `case`:
```haskell
case nodeOptions of
  SpoNodeOptions -> pure ()
  RelayNodeOptions -> do
    ...
```
Additionally, if you have a dedicated function, you can compose it.